aws/resource_aws_worklink_fleet.go

```diff
@@ -0,0 +1,399 @@
+	conn := meta.(*AWSClient).worklinkconn
+
+	input := &worklink.CreateFleetInput{
+		FleetName:                  aws.String(d.Get("name").(string)),
+		OptimizeForEndUserLocation: aws.Bool(d.Get("optimize_for_end_user_location").(bool)),
+	}
+
+	if v, ok := d.GetOk("display_name"); ok {
+		input.DisplayName = aws.String(v.(string))
+	}
+
+	resp, err := conn.CreateFleet(input)
+	if err != nil {
+		return fmt.Errorf("Error creating Worklink Fleet: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.FleetArn))
+	return resourceAwsWorkLinkFleetUpdate(d, meta)
+}
+
+func resourceAwsWorkLinkFleetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).worklinkconn
+
+	resp, err := conn.DescribeFleetMetadata(&worklink.DescribeFleetMetadataInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+			log.Printf("[WARN] Worklink Fleet (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error describe worklink fleet: %s", err)
+	}
+
+	d.Set("arn", d.Id())
+	d.Set("name", aws.StringValue(resp.FleetName))
+	d.Set("display_name", aws.StringValue(resp.DisplayName))
+	d.Set("optimize_for_end_user_location", aws.BoolValue(resp.OptimizeForEndUserLocation))
+	d.Set("company_code", aws.StringValue(resp.CompanyCode))
+	d.Set("created_time", resp.CreatedTime.Format(time.RFC3339))
+	if resp.LastUpdatedTime != nil {
+		d.Set("last_updated_time", resp.LastUpdatedTime.Format(time.RFC3339))
+	}
+	auditStreamConfigurationResp, err := conn.DescribeAuditStreamConfiguration(&worklink.DescribeAuditStreamConfigurationInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		if !isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
+			return fmt.Errorf("Error describe worklink audit stream configuration: %s", err)
+		}
+	}
+	d.Set("audit_stream_arn", auditStreamConfigurationResp.AuditStreamArn)
+
+	companyNetworkConfigurationResp, err := conn.DescribeCompanyNetworkConfiguration(&worklink.DescribeCompanyNetworkConfigurationInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		return fmt.Errorf("Error describe worklink company network configuration: %s", err)
+	}
+	d.Set("network", flattenWorkLinkNetworkConfigResponse(companyNetworkConfigurationResp))
+
+	identityProviderConfigurationResp, err := conn.DescribeIdentityProviderConfiguration(&worklink.DescribeIdentityProviderConfigurationInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		return fmt.Errorf("Error describe worklink company network configuration: %s", err)
+	}
+	d.Set("identity_provider", flattenWorkLinkIdentityProviderConfigResponse(identityProviderConfigurationResp))
+
+	devicePolicyConfigurationResp, err := conn.DescribeDevicePolicyConfiguration(&worklink.DescribeDevicePolicyConfigurationInput{
+		FleetArn: aws.String(d.Id()),
+	})
+	if err != nil {
+		if !isAWSErr(err, worklink.ErrCodeResourceNotFoundException, "") {
```

Comment:
If the error is `ErrCodeResourceNotFoundException` what value does `devicePolicyConfigurationResp` have? The below code is assuming it is not `nil` by dereferencing it, which will cause a panic if it is `nil`.

Reply:
Fix to just return err.